scene/gui/tree.cpp

```diff
@@ -3332,21 +3318,33 @@ void Tree::gui_input(const Ref<InputEvent> &p_event) {
 						drag_touching_deaccel = true;
 					}
 				}
-				update();
 			}
+
+			if (cache.click_type == Cache::CLICK_BUTTON && cache.click_item != nullptr) {
+				// make sure in case of wrong reference after reconstructing whole TreeItems
```

Comment:
```suggestion
				// Make sure in case of wrong reference after reconstructing whole TreeItems.
```

Reply:
This block was moved and that comment was already there. I can change it as you suggested if you want.